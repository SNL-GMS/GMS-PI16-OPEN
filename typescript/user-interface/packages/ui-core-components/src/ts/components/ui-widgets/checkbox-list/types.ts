import { Position } from '@blueprintjs/core';
import { IconName } from '@blueprintjs/icons';
import Immutable from 'immutable';

export interface CheckboxListProps {
  checkboxEnum: any;
  enumToCheckedMap: Map<any, boolean>;
  enumKeysToDisplayStrings?: Map<string, string>;
  enumToColorMap?: Map<any, string>;
  onChange(map: Map<any, boolean>);
}

export interface SimpleCheckboxListProps {
  checkBoxListEntries: CheckboxListEntry[];
  onChange?(name: string): void;
}

export interface SimpleCheckboxListState {
  checkboxEntriesMap: Immutable.Map<string, CheckboxListEntry>;
}

export interface CheckboxListEntry {
  name: string;
  color?: string;
  iconButton?: IconButton;
  isChecked: boolean;
}

export interface IconButton {
  iconName: IconName;
  popover?: {
    position: Position;
    usePortal?: boolean;
    minimal?: boolean;
    content?: string | JSX.Element;
  };
  onClick?(event: React.MouseEvent<HTMLButtonElement>): void;
}
