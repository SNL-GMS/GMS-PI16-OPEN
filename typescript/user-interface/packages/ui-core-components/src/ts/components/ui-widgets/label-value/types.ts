import React from 'react';

/**
 * app determines the default styling for the label-value toolbar item, defaults to soh
 */
export interface LabelValueProps {
  value: string;
  label: string;
  tooltip: string;
  valueColor?: string;
  ianApp?: boolean;
  styleForValue?: React.CSSProperties;
  containerClass?: string;
}
