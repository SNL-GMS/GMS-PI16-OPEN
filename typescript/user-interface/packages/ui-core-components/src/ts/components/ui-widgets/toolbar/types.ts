import { IconName } from '@blueprintjs/core';
import { DateTimeFormat } from '@gms/common-util';
import React from 'react';

import { DateRangePickerTypes } from '../date-range-picker';

export interface ToolbarProps {
  items: ToolbarItem[];
  toolbarWidthPx: number;
  itemsLeft?: ToolbarItem[];
  minWhiteSpacePx?: number;
  spaceBetweenItemsPx?: number;
  hidden?: boolean;
  overflowIcon?: IconName;
}
export interface ToolbarState {
  indicesToOverflow: number[];
  leftIndicesToOverflow: number[];
  whiteSpaceAllotmentPx: number;
  checkSizeOnNextDidMountOrDidUpdate: boolean;
}

export type ToolbarItem =
  | NumericInputItem
  | DropdownItem
  | IntervalPickerItem
  | DateRangePickerItem
  | PopoverItem
  | SwitchItem
  | ButtonItem
  | ButtonGroupItem
  | LabelValueItem
  | CheckboxDropdownItem
  | LoadingSpinnerItem
  | CustomItem;

interface ToolbarItemBase {
  label?: string;
  tooltip: string;
  tooltipForIssue?: string;
  type: ToolbarItemType;
  rank: number;
  style?: React.CSSProperties;
  widthPx?: number;
  labelRight?: string;
  disabled?: boolean;
  icon?: IconName;
  onlyShowIcon?: boolean;
  menuLabel?: string;
  cyData?: string;
  hasIssue?: boolean;
  ianApp?: boolean; // If true, use default ian styling. defaults to false (SoH)
  onMouseEnter?(): void;
  onMouseOut?(): void;
}

interface NumericInputItem extends ToolbarItemBase {
  value: number;
  minMax: MinMax;
  step?: number;
  requireEnterForOnChange?: boolean;
  onChange(value: number);
}

interface DropdownItem extends ToolbarItemBase {
  dropdownOptions: any;
  dropdownText?: any;
  value: any;
  custom?: boolean;
  displayLabel?: boolean;
  onChange(value: any);
}

interface IntervalPickerItem extends ToolbarItemBase {
  startDate: Date;
  endDate: Date;
  defaultIntervalInHours: number;
  shortFormat?: boolean;
  onChange(startDate: Date, endDate: Date);
  onApplyButton(startDate: Date, endDate: Date);
}

interface DateRangePickerItem extends ToolbarItemBase {
  startTimeMs: number;
  endTimeMs: number;
  format: DateTimeFormat;
  durations?: DateRangePickerTypes.DurationOption[];
  minStartTimeMs?: number;
  maxEndTimeMs?: number;
  onChange(startTimeMs: number, endTimeMs: number);
  onApplyButton(startTimeMs: number, endTimeMs: number);
}

interface PopoverItem extends ToolbarItemBase {
  popoverContent: JSX.Element;
  onPopoverDismissed();
}

interface SwitchItem extends ToolbarItemBase {
  value: boolean;
  onChange(value: boolean);
}

interface ButtonItem extends ToolbarItemBase {
  onClick();
}

interface ButtonGroupItem extends ToolbarItemBase {
  buttons: ButtonItem[];
}

interface LabelValueItem extends ToolbarItemBase {
  value: string;
  valueColor?: string;
  styleForValue?: React.CSSProperties;
}

interface CheckboxDropdownItem extends ToolbarItemBase {
  values: Map<any, boolean>;
  colors?: Map<any, string>;
  enumKeysToDisplayStrings?: Map<string, string>;
  enumOfKeys: any;
  onChange(value: any): void;
  onPopUp?(ref?: HTMLDivElement): void;
  onPopoverDismissed?(): void;
}

interface LoadingSpinnerItem extends ToolbarItemBase {
  itemsToLoad: number;
  itemsLoaded?: number;
  hideTheWordLoading?: boolean;
  hideOutstandingCount?: boolean;
}

interface CustomItem extends ToolbarItemBase {
  element: JSX.Element;
}

export interface MinMax {
  min: number;
  max: number;
}

export enum ToolbarItemType {
  Switch = 'Switch',
  Popover = 'Popover',
  Dropdown = 'Dropdown',
  NumericInput = 'NumericInput',
  Button = 'Button',
  IntervalPicker = 'IntervalPicker',
  DateRangePicker = 'DateRangePicker',
  ButtonGroup = 'ButtonGroup',
  LabelValue = 'LabelValue',
  CheckboxList = 'CheckboxList',
  LoadingSpinner = 'LoadingSpinner',
  CustomItem = 'CustomItem'
}

export {
  ButtonGroupItem,
  ButtonItem,
  CheckboxDropdownItem,
  CustomItem,
  DateRangePickerItem,
  DateRangePickerTypes,
  DropdownItem,
  IntervalPickerItem,
  LabelValueItem,
  LoadingSpinnerItem,
  NumericInputItem,
  PopoverItem,
  SwitchItem
};
