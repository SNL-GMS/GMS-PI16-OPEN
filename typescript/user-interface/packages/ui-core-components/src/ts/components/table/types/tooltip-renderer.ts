/* eslint-disable @typescript-eslint/no-empty-interface */
import { ITooltipParams as AgITooltipParams } from 'ag-grid-community';

/** Wrapper interface class around ag-grid interface `ITooltipParams` */
export interface TooltipParams extends AgITooltipParams {}
