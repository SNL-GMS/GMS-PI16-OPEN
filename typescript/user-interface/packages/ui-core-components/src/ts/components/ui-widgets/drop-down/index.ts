// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { DropDown } from './drop-down';

export { DropDown };
