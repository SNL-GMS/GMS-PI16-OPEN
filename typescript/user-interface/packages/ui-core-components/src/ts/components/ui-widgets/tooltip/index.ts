import { TooltipWrapper, WithTooltip } from './tooltip';

export { TooltipWrapper, WithTooltip };
