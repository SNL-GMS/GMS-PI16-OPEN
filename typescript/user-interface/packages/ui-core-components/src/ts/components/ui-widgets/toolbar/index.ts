// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { ToolbarComponent } from './toolbar-component';
import * as ToolbarTypes from './types';

export { ToolbarComponent as Toolbar, ToolbarTypes };
