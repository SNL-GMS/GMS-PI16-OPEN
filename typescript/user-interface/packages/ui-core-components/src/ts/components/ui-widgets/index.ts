import { CheckboxListEntry } from './checkbox-list';
import { CheckboxSearchList, CheckboxSearchListTypes } from './checkbox-search-list';
import { DateRangePicker, DateRangePickerTypes } from './date-range-picker';
// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { DropDown } from './drop-down';
import { FilterableOptionList, FilterableOptionListTypes } from './filterable-option-list';
// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { Form, FormTypes } from './form';
import { HistoryList, HistoryListTypes } from './history-list';
import { IntervalPicker, IntervalPickerTypes } from './interval-picker';
import { LabelValue } from './label-value';
import { LoadingSpinner } from './loading-spinner';
import { PopoverButton } from './popover-button';
import { TextArea } from './text-area';
import { TimePicker, TimePickerTypes } from './time-picker';
import { TitleBar, TitleBarTypes } from './title-bar';
// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { Toolbar, ToolbarTypes } from './toolbar';
import { TooltipWrapper, WithTooltip } from './tooltip';
import { Widget, WidgetTypes } from './widgets';

export { CheckboxSearchList, CheckboxSearchListTypes };
export { CheckboxListEntry };
export { SimpleCheckboxList } from './checkbox-list/simple-checkbox-list';
export { DropDown };
export { FilterableOptionList, FilterableOptionListTypes };
export { Form, FormTypes };
export { PopoverButton };
export { HistoryList, HistoryListTypes };
export { TimePicker, TimePickerTypes };
export { DateRangePicker, DateRangePickerTypes };
export { IntervalPicker, IntervalPickerTypes };
export { LabelValue };
export { LoadingSpinner };
export { TextArea };
export { Toolbar, ToolbarTypes };
export { TitleBar, TitleBarTypes };
export { Widget, WidgetTypes };
export { TooltipWrapper, WithTooltip };
