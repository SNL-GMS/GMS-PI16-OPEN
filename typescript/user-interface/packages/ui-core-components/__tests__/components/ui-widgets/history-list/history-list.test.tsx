/* eslint-disable react/jsx-props-no-spreading */
import * as Enzyme from 'enzyme';
import * as React from 'react';

import { HistoryList } from '../../../../src/ts/components/ui-widgets/history-list';
import * as HistoryListTypes from '../../../../src/ts/components/ui-widgets/history-list/types';

const historyListItem: HistoryListTypes.HistoryListItem = {
  index: 0,
  label: 'item',
  id: '1'
};

const props: HistoryListTypes.HistoryListProps = {
  items: [historyListItem],
  preferredItems: [historyListItem],
  listLength: 1,
  onSelect: jest.fn()
};

const wrapper = Enzyme.mount(<HistoryList {...props} />);

const props2: HistoryListTypes.HistoryListProps = {
  items: [historyListItem],
  preferredItems: [historyListItem],
  listLength: 1,
  onSelect: jest.fn()
};
const wrapper2 = Enzyme.mount(<HistoryList {...props2} />);

describe('History List', () => {
  it('to be defined', () => {
    expect(HistoryList).toBeDefined();
  });

  it('History List shallow renders', () => {
    const shallow = Enzyme.shallow(<HistoryList {...props} />);
    expect(shallow).toMatchSnapshot();
  });

  it('History List renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('History List shallow renders with other props', () => {
    const shallow = Enzyme.shallow(<HistoryList {...props2} />);
    expect(shallow).toMatchSnapshot();
  });

  it('History List 2 renders', () => {
    expect(wrapper2).toMatchSnapshot();
  });

  it('History List functions and clicks', () => {
    const wrapper3 = Enzyme.mount(<HistoryList {...props} />);
    const instance: HistoryList = wrapper3.find(HistoryList).instance() as HistoryList;

    const spy = jest.spyOn(instance, 'componentDidMount');
    instance.componentDidMount();
    expect(spy).toHaveBeenCalled();

    const input = wrapper3.find('HistoryList');
    const spy3 = jest.spyOn(instance, 'render');
    input.simulate('click');
    instance.render();
    expect(spy3).toHaveBeenCalled();
  });
});
