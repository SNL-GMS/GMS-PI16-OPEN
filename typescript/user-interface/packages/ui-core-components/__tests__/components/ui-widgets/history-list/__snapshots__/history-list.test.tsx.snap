// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History List History List 2 renders 1`] = `
<HistoryList
  items={
    Array [
      Object {
        "id": "1",
        "index": 0,
        "label": "item",
      },
    ]
  }
  listLength={1}
  onSelect={[MockFunction]}
  preferredItems={
    Array [
      Object {
        "id": "1",
        "index": 0,
        "label": "item",
      },
    ]
  }
>
  <div>
    <Blueprint3.Menu>
      <ul
        className="bp3-menu"
      >
        <Blueprint3.MenuItem
          disabled={false}
          key="1"
          multiline={false}
          onClick={[Function]}
          popoverProps={Object {}}
          shouldDismissPopover={true}
          text="item"
        >
          <li
            className=""
          >
            <a
              className="bp3-menu-item bp3-popover-dismiss"
              onClick={[Function]}
            >
              <Blueprint3.Icon />
              <Blueprint3.Text
                className="bp3-fill"
                ellipsize={true}
              >
                <div
                  className="bp3-text-overflow-ellipsis bp3-fill"
                >
                  item
                </div>
              </Blueprint3.Text>
            </a>
          </li>
        </Blueprint3.MenuItem>
        <Blueprint3.MenuDivider>
          <li
            className="bp3-menu-divider"
          />
        </Blueprint3.MenuDivider>
        <Blueprint3.MenuItem
          disabled={false}
          key="1"
          multiline={false}
          onClick={[Function]}
          popoverProps={Object {}}
          shouldDismissPopover={true}
          text="item"
        >
          <li
            className=""
          >
            <a
              className="bp3-menu-item bp3-popover-dismiss"
              onClick={[Function]}
            >
              <Blueprint3.Icon />
              <Blueprint3.Text
                className="bp3-fill"
                ellipsize={true}
              >
                <div
                  className="bp3-text-overflow-ellipsis bp3-fill"
                >
                  item
                </div>
              </Blueprint3.Text>
            </a>
          </li>
        </Blueprint3.MenuItem>
      </ul>
    </Blueprint3.Menu>
  </div>
</HistoryList>
`;

exports[`History List History List renders 1`] = `
<HistoryList
  items={
    Array [
      Object {
        "id": "1",
        "index": 0,
        "label": "item",
      },
    ]
  }
  listLength={1}
  onSelect={[MockFunction]}
  preferredItems={
    Array [
      Object {
        "id": "1",
        "index": 0,
        "label": "item",
      },
    ]
  }
>
  <div>
    <Blueprint3.Menu>
      <ul
        className="bp3-menu"
      >
        <Blueprint3.MenuItem
          disabled={false}
          key="1"
          multiline={false}
          onClick={[Function]}
          popoverProps={Object {}}
          shouldDismissPopover={true}
          text="item"
        >
          <li
            className=""
          >
            <a
              className="bp3-menu-item bp3-popover-dismiss"
              onClick={[Function]}
            >
              <Blueprint3.Icon />
              <Blueprint3.Text
                className="bp3-fill"
                ellipsize={true}
              >
                <div
                  className="bp3-text-overflow-ellipsis bp3-fill"
                >
                  item
                </div>
              </Blueprint3.Text>
            </a>
          </li>
        </Blueprint3.MenuItem>
        <Blueprint3.MenuDivider>
          <li
            className="bp3-menu-divider"
          />
        </Blueprint3.MenuDivider>
        <Blueprint3.MenuItem
          disabled={false}
          key="1"
          multiline={false}
          onClick={[Function]}
          popoverProps={Object {}}
          shouldDismissPopover={true}
          text="item"
        >
          <li
            className=""
          >
            <a
              className="bp3-menu-item bp3-popover-dismiss"
              onClick={[Function]}
            >
              <Blueprint3.Icon />
              <Blueprint3.Text
                className="bp3-fill"
                ellipsize={true}
              >
                <div
                  className="bp3-text-overflow-ellipsis bp3-fill"
                >
                  item
                </div>
              </Blueprint3.Text>
            </a>
          </li>
        </Blueprint3.MenuItem>
      </ul>
    </Blueprint3.Menu>
  </div>
</HistoryList>
`;

exports[`History List History List shallow renders 1`] = `
<div>
  <Blueprint3.Menu>
    <Blueprint3.MenuItem
      disabled={false}
      key="1"
      multiline={false}
      onClick={[Function]}
      popoverProps={Object {}}
      shouldDismissPopover={true}
      text="item"
    />
    <Blueprint3.MenuDivider />
    <Blueprint3.MenuItem
      disabled={false}
      key="1"
      multiline={false}
      onClick={[Function]}
      popoverProps={Object {}}
      shouldDismissPopover={true}
      text="item"
    />
  </Blueprint3.Menu>
</div>
`;

exports[`History List History List shallow renders with other props 1`] = `
<div>
  <Blueprint3.Menu>
    <Blueprint3.MenuItem
      disabled={false}
      key="1"
      multiline={false}
      onClick={[Function]}
      popoverProps={Object {}}
      shouldDismissPopover={true}
      text="item"
    />
    <Blueprint3.MenuDivider />
    <Blueprint3.MenuItem
      disabled={false}
      key="1"
      multiline={false}
      onClick={[Function]}
      popoverProps={Object {}}
      shouldDismissPopover={true}
      text="item"
    />
  </Blueprint3.Menu>
</div>
`;
